### PR TITLE
Introduce transaction generic helpers

### PR DESCRIPTION
**Pull request by fbiville**
Relates to #373

### CODE REVIEW

File: neo4j/transaction_helpers_example_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	_, err = result.Consume(ctx)
+	return err
+}
+
+func createDriver() (neo4j.DriverWithContext, error) {
+	credentials := neo4j.BasicAuth(os.Getenv("USERNAME"), os.Getenv("PASSWORD"), "")
+	return neo4j.NewDriverWithContext(os.Getenv("URL"), credentials)
+}
+
+type ctxCloser interface {
```
Comment (robsdedude) [inline neo4j/transaction_helpers_example_test.go:173]: nit: maybe `ctxCloseable` as it's not really closing the context.

File: neo4j/transaction_helpers.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func transactionWorkAdapter[T any](work ManagedTransactionWorkT[T]) ManagedTransactionWork {
+	return func(tx ManagedTransaction) (any, error) {
+		return work(tx)
+	}
+}
+
+// castGeneric performs a type assertion on the given `result` to the generic type T, unless an error has occurred.
+//
+// Implementation note: the function currently assumes that `result` is compatible with T and does not perform a soft
+// assertions.
```
Comment (robsdedude) [inline neo4j/transaction_helpers.go]:
```suggestion
// assertion.
```

**Review by robsdedude — APPROVED** [review]
What generic beauty :sparkles:

File: neo4j/transaction_helpers_example_test.go (hunk 1 above)
Comment (fbiville) [reply neo4j/transaction_helpers_example_test.go:173]: I picked `Closer` because of https://pkg.go.dev/io#Closer

**Review by bigmontz — APPROVED** [review]
🎨

**Pull request merged**